src/main/java/org/generationcp/middleware/dao/LocationDAO.java

```diff
@@ -281,6 +282,33 @@ public long countByCountry(final Country country) {
 		return new ArrayList<>();
 	}
 
+	public List<Location> getFilteredLocations(final Set<Integer> types, final List<Integer> locationIds, final String programUUID) {
+		final List<Location> locations;
+		try {
+
+			final Criteria criteria = this.getSession().createCriteria(Location.class);
+
+			if (types != null && !types.isEmpty()) {
+				criteria.add(Restrictions.in(LocationDAO.LTYPE, types));
+			}
+
+			if (locationIds != null && !locationIds.isEmpty()) {
+				criteria.add(Restrictions.in(LocationDAO.LOCID, locationIds));
+			}
+
+			criteria.add(
+				Restrictions.or(Restrictions.eq(LocationDAO.UNIQUE_ID, programUUID), Restrictions.isNull(LocationDAO.UNIQUE_ID)));
```

Comment:
@cuenyad, does it work if programUUID is null?

Reply:
@clarysabel the programUUID is mandatory for the service and the condition is to recover only the locations that have the same programUUID or the programUUID is null this is because you can create locations for a program.

Reply:
@clarysabel I verified and if the programUUID is null the query continues working properly.